### DISCUSSION TITLE
Blog 6: Funktionierender Chat

### DISCUSSION THREAD

**Discussion by Max-Rohrhirsch** [DHBW-UMOC/UMOC#10] [Announcements]
# Zweiter Prototyp
Das UMOC Team hat den Messenger die letzte Woche um einige wichtige Schritte vorangebracht. Zum einen hat der Messenger ein Frontend erhalten, mit dem es nun erstmals möglich ist Nachrichten untereinander Auszutauschen (siehe Screenshot 1). Zum anderen wurde das Backend um weitere wichtige Schnittstellen erweitert, welche unter anderem das Anzeigen aktiver Chats und deren dazugehörigen Nachrichten ermöglichen (genaueres im Kap. Schnittstellen im Backend). Zum ersten mal fühlt sich das Projekt nun wie ein richtiger Messenger an, und ermöglicht einen Ausblick auf weitere Features die in den nächsten Wochen und Monaten folgen werden.

![image](https://github.com/user-attachments/assets/4662810b-88dd-4eef-8be2-cb7f4f250941)

# Schnittstellen im Backend 
Folgende Schnittstellen sind bereits implementiert und nutzen die Datenbank. Jedoch ist noch keine Verschlüsselung implementiert.

REST-API Endpoints
1. Register User
Endpoint: /register
Method: POST
Parameters:

username (string, required): The username for the new user.
password (string, required): The password for the new user.
public_key (string, optional, currently unused).

2. Login User
Endpoint: /login
Method: GET
Parameters:

username (string, required): The username of the user.
password (string, required): The password of the user.

3. Logout User
Endpoint: /logout
Method: POST
Parameters:

sessionID (string, required): The session ID of the user.

4. Add Contact
Endpoint: /addContact
Method: POST
Parameters:

sessionID (string, required): The session ID of the user.
contact (string, required): The contact's username.

5. Change Contact Status
Endpoint: /changeContact
Method: POST
Parameters:

sessionID (string, required): The session ID of the user.
contact (string, required): The contact's username.
status (string, required): The new status of the contact.

6. Get Contacts
Endpoint: /getContacts
Method: GET
Parameters:

sessionID (string, required): The session ID of the user.

7. Get Contact Messages
Endpoint: /getContactMessages
Method: GET
Parameters:

sessionID (string, required): The session ID of the user.
contact (string, required): The contact's username.

8. Default Test Endpoint
Endpoint: /
Method: GET

Es wurden auch Websocket Endpoints hinzugefügt, jedoch noch nicht getestet, weshalb sie in einem zukünftigem Blog erklärt werden.

**Comment by bunki1224** [comment]
Sieht nach viel Fortschritt diese Woche aus. Besonders gut gefällt mir der erste Prototyp des Frontend, man kann auf jeden Fall erkennen wo die Reise hingeht. Auch im Backend habt ihr wichtige Fortschritte gemacht. Ich bin auf den zukünftigen Blog gespannt, in dem er die einzelnen Funktionen der Endpoints erklärt. 

Ihr habt neben der Textleiste auch ein Plus Icon. Soll es später möglich sein Gifs / Bilder / Videos zu versenden?

>MFG TripAhead Team (JB)

**Comment by DonKir15** [comment]
Wow, ihr habt echt ordentlich was geschafft diese Woche! Der erste Prototyp vom Frontend sieht schon super aus – man erkennt direkt, dass es jetzt wie ein echter Messenger wirkt. Dass ihr Nachrichten schon austauschen könnt, ist ein großer Schritt!
Auch das Backend macht ordentlich Fortschritte. Die Liste der Endpoints zeigt, dass ihr eine solide Grundlage legt, auch wenn die Verschlüsselung noch fehlt. Bin gespannt, wie es weitergeht und was für Features ihr noch plant. Freue mich schon auf den nächsten Blog, wo ihr hoffentlich mehr Details zu den Funktionen bringt. Super Arbeit, weiter so!
